Custom Error type is not assignable to variable of type error

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

type CustomError string

func (s CustomError) Error() string {
        return string(s)
}

func NewCustomError(errorText string) CustomError {
        return CustomError(errorText)
}

func main() {
        var myError error
        myError = NewCustomError("Everything is going wrong!")
        fmt.Println(myError)
}
```

Expected result:
```console
$ go run ./sample.go
Everything is going wrong!
```

Got:
```console
$ yaegi ./sample.go
./sample.go:12:9: panic
./sample.go:17:12: panic
run: reflect.Set: value of type string is not assignable to type error
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000544400, 0x40)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc000551c18)
        /go/src/yaegi/interp/interp.go:503 +0xc8
panic(0x13bf140, 0xc000114b50)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc0000c82c0, 0xc000547400, 0xc0005519f8)
        /go/src/yaegi/interp/run.go:185 +0x253
panic(0x13bf140, 0xc000114b50)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc0000c8370, 0xc000546900, 0xc0005517e8)
        /go/src/yaegi/interp/run.go:185 +0x253
panic(0x13bf140, 0xc000114b50)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
reflect.Value.assignTo(0x13bf140, 0xc000114b30, 0x198, 0x1533584, 0xb, 0x1423b00, 0xc000114b10, 0xc000114b40, 0x98, 0xc000114b40)
        /usr/local/go/src/reflect/value.go:2451 +0x3f7
reflect.Value.Set(0x1423b00, 0xc000114b10, 0x194, 0x13bf140, 0xc000114b30, 0x198)
        /usr/local/go/src/reflect/value.go:1564 +0xbd
github.com/traefik/yaegi/interp._return.func1(0xc0000c8370, 0xc000114a70)
        /go/src/yaegi/interp/run.go:2274 +0x85
github.com/traefik/yaegi/interp.runCfg(0xc000546900, 0xc0000c8370)
        /go/src/yaegi/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.call.func7(0xc0000c82c0, 0xc0004571f0)
        /go/src/yaegi/interp/run.go:1296 +0xd05
github.com/traefik/yaegi/interp.runCfg(0xc000547400, 0xc0000c82c0)
        /go/src/yaegi/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc00000c1e0, 0xc000546c00, 0xc0000c8d10)
        /go/src/yaegi/interp/run.go:122 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc00000c1e0, 0xc0000ac9a0, 0x14c, 0x7fff888d2ef7, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/yaegi/interp/interp.go:595 +0x80c
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc00000c1e0, 0x7fff888d2ef7, 0xb, 0xc0000ac800, 0x14c, 0x0, 0x0, 0xc0000afcb0)
        /go/src/yaegi/interp/interp.go:417 +0x105
main.runFile(0xc00000c1e0, 0x7fff888d2ef7, 0xb, 0x0, 0x1, 0x0)
        /go/src/yaegi/cmd/yaegi/run.go:139 +0xb1
main.run(0xc000032050, 0x1, 0x1, 0xc00000a501, 0xc000001680)
        /go/src/yaegi/cmd/yaegi/run.go:102 +0x99d
main.main()
        /go/src/yaegi/cmd/yaegi/yaegi.go:144 +0x415
```

Follow-up of https://github.com/traefik/yaegi/issues/1091

--- a/_test/ret8.go
+++ b/_test/ret8.go
@@ -18,7 +18,11 @@ func fail() (err error) {
 
 func main() {
 	fmt.Println(fail())
+	var myError error
+	myError = NewCustomError("ok")
+	fmt.Println(myError)
 }
 
 // Output:
 // Everything is going wrong!
+// ok

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -557,9 +557,6 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 				n.findex = dest.findex
 				n.level = dest.level
 
-				// Propagate type.
-				// TODO: Check that existing destination type matches source type.
-
 				// In the following, we attempt to optimize by skipping the assign
 				// operation and setting the source location directly to the destination
 				// location in the frame.

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -577,6 +577,10 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 					// Setting a struct field of function type requires an extra step. Do not optimize.
 				case isCall(src) && !isInterfaceSrc(dest.typ) && !isRecursiveField(dest) && n.kind != defineStmt:
 					// Call action may perform the assignment directly.
+					if dest.typ.id() != src.typ.id() {
+						// Skip optimitization if returned type doesn't match assigned one.
+						break
+					}
 					n.gen = nop
 					src.level = level
 					src.findex = dest.findex

